src/lib/services/OBRHelper.ts

```diff
@@ -52,12 +61,23 @@ function subscribeToRoomNotifications() {
 }
 
 async function initGM() {
+  GmId.set(OBR.player.id)
+  TrackedPlayer.set(OBR.player.id)
+
+  OBR.player.onChange((gm) => {
+    GmPlayer.set(gm);
+  });
+
   OBR.party.onChange((party) => {
     PartyStore.set(party);
   });
 
-  PartyStore.subscribe((party) => {
+  PartyStore.subscribe(async (party) => {
     const pmd: { [pId: string]: PlayerMetaData } = {};
+
+    // add GM metadata too
+    pmd[OBR.player.id] = (await OBR.player.getMetadata())[pluginId("sheetData")];
```

Comment:
This code only runs for the GM currently (this subscriber only exists if the uses ir the GM, since it subscribes inside `initGM()`) so this is already being done only for the GM currently.

If support for all players to see other players' sheets then I believe this would need to run for all players, but I think many other changes would also be needed (basically merging the code for `initGM` and `initPlayer` and refactor the code in terms of a `currentPlayer` instead of `GM`).

Reply:
gotcha